contracts/Core/BlockManager.sol

```diff
@@ -256,27 +256,37 @@ contract BlockManager is Initializable, ACL, BlockStorage, StateManager, IBlockM
             return;
         }
 
-        uint8 pushAt = uint8(sortedProposedBlockIds[epoch].length);
-        for (uint8 i = 0; i < sortedProposedBlockIds[epoch].length; i++) {
-            if (proposedBlocks[epoch][sortedProposedBlockIds[epoch][i]].biggestInfluence < biggestInfluence) {
-                pushAt = i;
+        for (uint8 i = 0; i <= sortedProposedBlockIds[epoch].length; i++) {
+            // Push at last, have worst itr
+            if (i == sortedProposedBlockIds[epoch].length && sortedProposedBlockIds[epoch].length < parameters.maxAltBlocks()) {
+                sortedProposedBlockIds[epoch].push(blockId);
                 break;
             }
-            if (proposedBlocks[epoch][sortedProposedBlockIds[epoch][i]].iteration > iteration) {
-                pushAt = i;
+            // Replace : New Block has better biggest influence
+            else if (proposedBlocks[epoch][sortedProposedBlockIds[epoch][i]].biggestInfluence < biggestInfluence) {
```

Comment:
sortedProposedBlockIds[epoch][i] will cause index out of bounds when i==sortedProposedBlockIds[epoch].length

Reply:
hmm I thought that first if wont allow to reach here 
but it would reach once blocks > maxAltBlocks 
cool will resolve this

Reply:
true. but gas intensive to compare biggestinfluence with every block.
Not required if proposer can be slashed in dispute by proving higher biggestInfluence.
Which is missing and important.

Reply:
Fixed https://github.com/razor-network/contracts/pull/483/commits/226903825512b9087390fa4d297389f3502cccdb